README.md

```diff
@@ -138,6 +138,17 @@ form:
 }
 ```
 
+`for..of` construct:
+
+The bloomrun instance and iterator are all implements iterable protocol,
+patterns can be looped over in a `for..of` construct. Mentioned that, 
+the loop over bloomrun instance will iterate all available patterns in 
+bloomrun instance.
+
+```js
+for (var pattern of instance)
+```
```

Comment:
can you make this about the iterator itelself, and talk about main object at the top (after the constructor)?